### PR TITLE
fix: validate entries in nox.config, too, using attrs

### PR DESCRIPTION
**Pull request by henryiii**
This is a followup to #871. It also reduces the need for #872, as I think it makes it harder or impossible to incorrectly set this via normal means.

Attrs is a well respected, zero dependency, small library by a CPython core dev, but it is an extra dependency. If we did adopt it, though, I would start moving over a few other things, which could improve our internal typing too.

<details>
<summary>Incomplete list for "NoxCmdOptions":</summary>

```python
class NoxCmdOptions(NoxOptions):
    version: bool
    list_sessions: bool
    json: bool
    posargs: list[str]
    add_timestamp: bool
    no_venv: bool
    R: bool
    noxfile: str
    envdir: None | str
    extra_pythons: list[str]
    force_pythons: list[str]
    install_only: bool
    no_install: bool
    report: None | str
    no_reuse_existing_virtualenvs: bool  # backcompat
    non_interactive: bool
    nocolor: bool
    forcecolor: bool
    color: bool
    invoked_from: str
```

</details>

### CODE REVIEW

**Review by DiddiLeija — COMMENTED** [review]
Looks like a very handy fix to me! However I'd like to see what other mantainers think of adding a dependency before giving the approval :)

**Comment by henryiii** [comment]
It's been a couple of weeks without any complaints, and I'd like to build on this, so I think I'll merge unless there are any last objections!

**Review by Milad335t — APPROVED** [review]

**Review by Milad335t — APPROVED** [review]

**Review by Milad335t — APPROVED** [review]

**Pull request merged**
